tlsconfig/config_test.go

```diff
@@ -478,6 +486,42 @@ func TestConfigClientTLSValidClientCertAndKey(t *testing.T) {
 	}
 }
 
+// The certificate is set if the client cert and encrypted client key are
+// provided and valid and passphrase can decrypt the key
+func TestConfigClientTLSValidClientCertAndEncryptedKey(t *testing.T) {
+	key, cert := getCertAndEncryptedKey()
+
+	tlsConfig, err := Client(Options{
+		CertFile:   cert,
+		KeyFile:    key,
+		Passphrase: "FooBar123",
+	})
+
+	if err != nil || tlsConfig == nil {
+		t.Fatal("Unable to configure client TLS", err)
+	}
+
+	if len(tlsConfig.Certificates) != 1 {
+		t.Fatal("Unexpected client certificates")
+	}
+}
+
+// The certificate is not set if the provided passphrase cannot decrypt
+// the encrypted key.
+func TestConfigClientTLSNotSetWithInvalidPassphrase(t *testing.T) {
+	key, cert := getCertAndEncryptedKey()
+
+	tlsConfig, err := Client(Options{
+		CertFile:   cert,
+		KeyFile:    key,
+		Passphrase: "InvalidPassphrase",
+	})
+
+	if err == nil || tlsConfig != nil {
+		t.Fatal("Expected failure due to incorrect passphrase.")
+	}
```

Comment:
Thank you for the review @cyli. Yes, this seems a better test. I have updated the PR to address both changes (pem.EncodeToMemory() and the specific passphrase error in the test.